src/channeldef.ts

```diff
@@ -1072,14 +1116,12 @@ export function initFieldDef(fd: FieldDef<string, any>, channel: Channel) {
   } else if (!isSecondaryRangeChannel(channel)) {
     // If type is empty / invalid, then augment with default type
     const newType = defaultType(fieldDef as TypedFieldDef<any>, channel);
-    log.warn(log.message.missingFieldType(channel, newType));
-
     fieldDef['type'] = newType;
   }
 
   if (isTypedFieldDef(fieldDef)) {
-    const {compatible, warning} = channelCompatibility(fieldDef, channel);
-    if (!compatible) {
+    const {compatible, warning} = channelCompatibility(fieldDef, channel) || {};
```

Comment:
```suggestion
    const {compatible, warning} = channelCompatibility(fieldDef, channel) ?? {};
```

Reply:
Meh, It doesn't matter here and `||` leads to better ES5 output.